http-client/Network/HTTP/Client/Types.hs

```diff
@@ -115,12 +115,6 @@ data HttpException = StatusCodeException Status ResponseHeaders CookieJar
                    -- ^ Environment name and value
                    --
                    -- Since 0.4.7
-
-                   | ResponseLengthAndChunkingBothUsed
```

Comment:
We should leave this in for now, to avoid breaking code using it. We should also document that the exception will never be thrown.